80_transform_jobs_crud can fail with a null pointer exception

It looks like `Transform#transformTaskNodes` can sometimes produce a null node ID, which causes a null pointer exception in `DiscoveryNodes.resolveNodes`.

Reproduce line:
```
./gradlew ':x-pack:qa:rolling-upgrade:v7.8.0#oneThirdUpgradedTest' --tests "org.elasticsearch.upgrades.UpgradeClusterClientYamlTestSuiteIT.test {p0=mixed_cluster/80_transform_jobs_crud/Test GET, start, and stop old cluster batch transforms}" \
  -Dtests.seed=5F2B3A83763FEB92 \
  -Dtests.security.manager=true \
  -Dtests.locale=ga-IE \
  -Dtests.timezone=America/Phoenix \
  -Dcompiler.java=14 \
  -Druntime.java=11 \
  -Dtests.rest.suite=mixed_cluster
```

Build scan: https://gradle-enterprise.elastic.co/s/uwebhlksubme2

Relevant excerpt from logs:
```
  1> Caused by: java.lang.NullPointerException: Cannot invoke \"String.equals(Object)\" because \"nodeId\" is null
  1> 	at org.elasticsearch.cluster.node.DiscoveryNodes.resolveNodes(DiscoveryNodes.java:341)
  1> 	at org.elasticsearch.action.support.tasks.TransportTasksAction.resolveNodes(TransportTasksAction.java:159)
  1> 	at org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.<init>(TransportTasksAction.java:230)
  1> 	at org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.<init>(TransportTasksAction.java:215)
  1> 	at org.elasticsearch.action.support.tasks.TransportTasksAction.doExecute(TransportTasksAction.java:96)
  1> 	at org.elasticsearch.xpack.transform.action.TransportStopTransformAction.lambda$doExecute$2(TransportStopTransformAction.java:188)
  1> 	at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:63)
  1> 	at 
```

--- a/x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/TransformSurvivesUpgradeIT.java
+++ b/x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/TransformSurvivesUpgradeIT.java
@@ -123,7 +123,6 @@ public class TransformSurvivesUpgradeIT extends AbstractUpgradeTestCase {
      * The purpose of this test is to ensure that when a job is open through a rolling upgrade we upgrade the results
      * index mappings when it is assigned to an upgraded node even if no other ML endpoint is called after the upgrade
      */
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/56196")
     public void testTransformRollingUpgrade() throws Exception {
         assumeTrue("Continuous transform not supported until 7.3", UPGRADE_FROM_VERSION.onOrAfter(Version.V_7_3_0));
         Request adjustLoggingLevels = new Request("PUT", "/_cluster/settings");

--- a/x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/mixed_cluster/80_transform_jobs_crud.yml
+++ b/x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/mixed_cluster/80_transform_jobs_crud.yml
@@ -1,7 +1,3 @@
-setup:
-  - skip:
-      version: "all"
-      reason: "AwaitsFix https://github.com/elastic/elasticsearch/issues/56250"
 ---
 "Test put batch transform on mixed cluster":
   - skip:

--- a/x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/old_cluster/80_transform_jobs_crud.yml
+++ b/x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/old_cluster/80_transform_jobs_crud.yml
@@ -1,7 +1,3 @@
-setup:
-- skip:
-    version: "all"
-    reason: "AwaitsFix https://github.com/elastic/elasticsearch/issues/56250"
 ---
 "Test put batch transform on old cluster":
   - skip:

--- a/x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/upgraded_cluster/80_transform_jobs_crud.yml
+++ b/x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/upgraded_cluster/80_transform_jobs_crud.yml
@@ -1,7 +1,4 @@
 setup:
-  - skip:
-      version: "all"
-      reason: "AwaitsFix https://github.com/elastic/elasticsearch/issues/56250"
   - do:
       cluster.health:
         wait_for_status: green

